### DISCUSSION TITLE
Change color_input button background to match selected color

### DISCUSSION THREAD

**Discussion by adosikas** [zauberzeug/nicegui#1390] [Ideas / Feature Requests]
I find it extremely useful to color the button of color inputs in the matching color (and setting the icon color to either white or black depending on how dark the color is).

I would assume making this an option or even the default behavior could be very helpful as it allows a preview of the color without having to open the picker.

![grafik](https://github.com/zauberzeug/nicegui/assets/3721366/16718681-3e8e-45dd-a0fa-ad13b9c17057)

```py
class ColorableColorInput(ui.color_input):
    def on_value_change(self, value: Any) -> None:
        super().on_value_change(value)
        contrast_color = "white" if _check_if_dark(value) else "black"
        self.button.style(f"background-color:{value or 'white'}!important; color:{contrast_color}!important")
```

**Comment by rodja** [comment]
Cool! I think it should be a parameter: `ui.color_picker(preview=True)`. I'm also in favour of making the default `True`. Would you like to create a pull request for this? See [CONTRIBUTING.md](https://github.com/zauberzeug/nicegui/blob/main/CONTRIBUTING.md#pull-requests) for a short explanation.

**Reply by adosikas** [reply]
I would be interested, but I am not that experienced with CSS and could not find a "clean" way of determining the color brightness (to decide what the contrast color will be).
Ideally this would also be done in conjunction with #1389 and would therefore need to support not just `#rrggbb` and `rgb(r,g,b)`, but also named colors (and stuff like `primary`), and I have no idea how to approach that.

I am currently using `pydantic_extra_types.color.Color(value).as_hsl_tuple()[2] < 0.5` which covers a bunch, but I don't think that is an requirement already.

**Reply by falkoschindler** [reply]
Oh yes, the automatic adjustment of the foreground color is a tricky subject. But maybe we can use something like https://github.com/ubernostrum/webcolors or simply the list of CSS color names https://github.com/ubernostrum/webcolors/blob/trunk/src/webcolors/constants.py.

**Reply by adosikas** [reply]
![grafik](https://github.com/zauberzeug/nicegui/assets/3721366/e3447f42-18c4-4b64-bcc5-cc73504abb7c)

A workaround to the whole contrast thing would be just adding a border around the icon.

**Reply by falkoschindler** [reply]
Great idea, @adosikas! Do you have an idea in mind how to implement that?

**Reply by markospy** [reply]
In css there are some properties for it, such as filter:drop-shadow and filter:text-shadow

**Reply by adosikas** [reply]
I just used multiple `text-shadow` . See #1473

**Comment by falkoschindler** [comment]
This feature has been released in version 1.3.12.
